api/src/main/java/org/openmrs/validator/AllergyValidator.java

```diff
@@ -63,6 +63,11 @@ public void validate(Object target, Errors errors) {
 		ValidationUtils.rejectIfEmpty(errors, "patient", "allergyapi.patient.required");
 		
 		Allergy allergy = (Allergy) target;
+
+		    // Additional validation: Ensure allergen does not contain numeric values
+			if (allergy.getAllergen() != null && StringUtils.isNumeric(allergy.getAllergen().getNonCodedAllergen())) {
+				errors.rejectValue("allergen", "error.allergyapi.allergy.Allergen.cannotContainNumeric");
+			}
```

Comment:
It is better to move the `if` block inside the `else` block at line 79.

Reply:
@ManojLL Done. Please merge it.